### PR TITLE
Improve aliases for clickhouse binary

### PR DESCRIPTION
**Pull request by azat**
### Changelog category (leave one):
- Build/Testing/Packaging Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Improve aliases for clickhouse binary (now `ch`/`clickhouse` is `clickhouse-local` or `clickhouse` depends on the arguments) and add bash completion for new aliases.

Follow-up for: https://github.com/ClickHouse/ClickHouse/pull/56634 (cc @alexey-milovidov )
_P.S. changelog entry copied from https://github.com/ClickHouse/ClickHouse/pull/56634_

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
*This is an automated comment for commit 5394426134302334adddea0febaec2c6350bb847 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/58344/5394426134302334adddea0febaec2c6350bb847/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/58344/5394426134302334adddea0febaec2c6350bb847/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: programs/main.cpp [hunk 1]
```diff
@@ -502,6 +501,11 @@ int main(int argc_, char ** argv_)
         }
     }
 
+    /// Interpret binary without argument or with arguments starts with dash
+    /// ('-') as clickhouse-local for better usability
```
[the remark this answers is not in the document]
Comment (azat) [reply programs/main.cpp:505]: It looks useful to me, i.e.

```sh
clickhouse # dumps help
clickhouse -q 'select 1' # use local
clickhouse # spawn local
clickhouse local # spawn local
```

Suggestions?

[the remark this answers is not in the document]
Comment (alexey-milovidov) [reply programs/main.cpp:505]: Good, I didn't understand it.

**Review by alexey-milovidov — APPROVED** [review]

File: programs/main.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (azat) [reply programs/main.cpp]: Ok, done.

**Comment by azat** [comment]
>Stateless tests (release, analyzer) — fail: 1, passed: 5963, skipped: 8

- https://github.com/ClickHouse/ClickHouse/pull/58352

**Pull request merged**